status/reporters.go

```diff
@@ -413,3 +514,145 @@ func (r *GitHubReporter) ReportStatus(k8sClient client.Client, ctx context.Conte
 
 	return nil
 }
+
+// ReportStatusForSnapshot creates CheckRun when using GitHub App integration.
+// create a a commit status when using GitHub webhook integration
+func (r *GitHubReporter) ReportStatusForSnapshot(k8sClient client.Client, ctx context.Context, logger *helpers.IntegrationLogger, snapshot *applicationapiv1alpha1.Snapshot, details *[]gitops.IntegrationTestStatusDetail) error {
+	if !metadata.HasLabelWithValue(snapshot, gitops.PipelineAsCodeEventTypeLabel, gitops.PipelineAsCodePullRequestType) {
+		return nil
+	}
+
+	labels := snapshot.GetLabels()
+
+	owner, found := labels[gitops.PipelineAsCodeURLOrgLabel]
+	if !found {
+		return fmt.Errorf("snapshot label not found %q", gitops.PipelineAsCodeURLOrgLabel)
+	}
+
+	repo, found := labels[gitops.PipelineAsCodeURLRepositoryLabel]
+	if !found {
+		return fmt.Errorf("snapshot label not found %q", gitops.PipelineAsCodeURLRepositoryLabel)
+	}
+
+	sha, found := labels[gitops.PipelineAsCodeSHALabel]
+	if !found {
+		return fmt.Errorf("snapshot label not found %q", gitops.PipelineAsCodeSHALabel)
+	}
+	// Existence of the Pipelines as Code installation ID annotation signals configuration using GitHub App integration.
+	// If it doesn't exist, GitHub webhook integration is configured.
+	if metadata.HasAnnotation(snapshot, gitops.PipelineAsCodeInstallationIDAnnotation) {
+		creds, err := r.getAppCredentials(ctx, snapshot)
+		if err != nil {
+			logger.Error(err, "failed to get app credentials from Snapshot",
+				"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
+			return err
+		}
+
+		token, err := r.client.CreateAppInstallationToken(ctx, creds.AppID, creds.InstallationID, creds.PrivateKey)
+		if err != nil {
+			logger.Error(err, "failed to create app installation token",
+				"creds.AppID", creds.AppID, "creds.InstallationID", creds.InstallationID)
+			return err
+		}
+
+		r.client.SetOAuthToken(ctx, token)
+
+		allCheckRuns, err := r.client.GetAllCheckRunsForRef(ctx, owner, repo, sha, creds.AppID)
+		if err != nil {
+			logger.Error(err, "failed to get all checkruns for ref",
+				"owner", owner, "repo", repo, "creds.AppID", creds.AppID)
+			return err
+		}
+
+		for _, integrationTestStatusDetail := range *details {
+			// We report the final status of handling a integrationTestScenario only
+			// such as TestFail, TestPassed, DeploymentError, EnvironmentProvisionError
+			integrationTestStatusDetail := integrationTestStatusDetail // G601
+			if integrationTestStatusDetail.Status == gitops.IntegrationTestStatusEnvironmentProvisionError || integrationTestStatusDetail.Status == gitops.IntegrationTestStatusDeploymentError || integrationTestStatusDetail.Status == gitops.IntegrationTestStatusTestFail || integrationTestStatusDetail.Status == gitops.IntegrationTestStatusTestPassed {
```

Comment:
https://github.com/redhat-appstudio/integration-service/pull/318 this could improve readability here

Reply:
Why we actually report only final status?

Reply:
At first, I decide to report the final result. Thanks for finding. I updated to report all statuses.